src/scala/org/pantsbuild/scoverage/report/ScoverageReport.scala

```diff
@@ -0,0 +1,254 @@
+package org.pantsbuild.scoverage.report
+
+import java.io.File
+import sbt.internal.util.ConsoleLogger
+import org.apache.commons.io.FileUtils
+
+import scoverage.{ Coverage, IOUtils, Serializer }
+import scoverage.report.{ ScoverageHtmlWriter, ScoverageXmlWriter }
+
+object ScoverageReport {
+  val Scoverage = "scoverage"
+
+  // Copying the logger from org.pantsbuild.zinc.bootstrap
+  // As per https://github.com/pantsbuild/pants/issues/6160, this is a workaround
+  // so we can run zinc without $PATH (as needed in remoting).
+  System.setProperty("sbt.log.format", "true")
+
+  // Setting the logger
+  val logger = ConsoleLogger.apply()
+
+  case class ReportOptions(
+    loadDataDir: Boolean,
+    measurementsPath: String,
+    sourcePath: String,
+    reportPath: String,
+    dataPath: String,
+    writeHtml: Boolean,
+    writeXml: Boolean,
+    writeXmlDebug: Boolean,
+    cleanOld: Boolean,
+    targetFilters: Seq[String]) {
+    val writeReports: Boolean = {
+      val write = writeHtml || writeXml
+      if (!write) {
+        // we could end here, but we will still attempt to load the coverage and dump stats as info
+        // to the log, so maybe somebody would want to run this w/o generating reports?
+        logger.warn("No report output format specified, so no reports will be written.")
+      }
+      write
+    }
+  }
+
+  object ReportOptions {
+    def apply(settings: Settings): ReportOptions = {
+      ReportOptions(
+        loadDataDir = settings.loadDataDir,
+        measurementsPath = settings.measurementsDirPath,
+        sourcePath = settings.sourceDirPath,
+        reportPath = settings.reportDirPath,
+        dataPath = settings.dataDirPath,
+        writeHtml = settings.writeHtmlReport,
+        writeXml = settings.writeXmlReport,
+        writeXmlDebug = settings.writeXmlDebug,
+        cleanOld = settings.cleanOldReports,
+        targetFilters = settings.targetFilters)
+    }
+  }
+  /**
+   *
+   * @param dataDirs list of measurement directories for which coverage
+   *                 report has to be generated
+   * @return         [Coverage] object for the [dataDirs]
+   */
+  private def aggregatedCoverage(dataDirs: Seq[File]): Coverage = {
+    var id = 0
```

Comment:
Possibly could use AtomicInteger here if trying to increment ids e.g., though this class is more for thread-safe op.

```
import java.util.concurrent.atomic.AtomicInteger
val id = new AtomicInteger()
...
coverage add stmt.copy(id = id.incrementAndGet())
```